build.gradle.kts

```diff
@@ -1,214 +1,259 @@
+        from(sharedManifest)
+    }
+    dependsOn("copyHelp")
+}
+
+tasks.assemble{
+    dependsOn("jar") 
+}
+
+// NoHelp JAR
+tasks.create<Jar>("noHelpJar"){
+    classifier = "nohelp"
+    dependsOn(configurations.runtimeClasspath)
+    from(sourceSets.main.get().output)
+    from({
+        configurations.runtimeClasspath.get().filter { it.name.contains("picocli") && it.name.endsWith("jar") }.map { zipTree(it) }
+    })
+    manifest {
+        attributes["Main-Class"] = application.mainClassName
+        from(sharedManifest)
+    }
+}
+
+/*
+ * Code coverage report tasks
+ */
+tasks.jacocoTestReport {
+    reports {
+        xml.isEnabled = true
+        csv.isEnabled = false
+        html.isEnabled = false
+    }
+}
+
+tasks.check{
+    dependsOn("jacocoTestReport")
+}
+
+tasks.register("release") {
+    group = "Release"
+    description = "Creates all artifacts for a given EduMIPS64 release"
+    dependsOn("allDocs")
+    dependsOn("jar")
+
+    doFirst {
+        println("Creating artifacts for version $version")
+    }
+}
+
+tasks.create<Exec>("exeCreator"){
+    group = "Distribution"
+    description = "Creates an installable exe file"
+    workingDir = File("${projectDir}")
+    dependsOn("jar")
+
+    doFirst{
+    	//check if user is on Windows
+        var os = System.getProperty("os.name") as String;
+        if(!("Windows" in os)){
+            throw GradleException("This tasks shall run on Windows")
+        }
+
+        println("Creating edumips64-"+version+".jar File");
+        
+        //check if jdk 14+ is installed
+        if(System.getProperty("java.version").toInt()<14) {
+            throw GradleException("JDK version installed must be 14+")
+        }
+        else{
+            println("Correct Jdk version Installed!")
+        }
+        
+        //check if Wix is installed
+        if(System.getenv("WIX")==null){
+            throw GradleException("Please Install Wix")
+        }
+        else{
+            println("Wix Installed!")
+        }
+
+	//create script file that creates the executable
+        var script = File("./scripts/exeCreator.bat")
+        script.createNewFile()
+        var version_arr = version.split('.').toMutableList();
+        if(version_arr.size>3 && version_arr[3].toInt()>0) {
+            version_arr[2] = (version_arr[2].toInt()+ 1).toString()
+        }
+        var jpackage_version = version_arr[0]+'.'+version_arr[1]+'.'+version_arr[2]
+        println("Running JPackage!")
+        script.writeText("jpackage.exe --main-jar edumips64-"+version+".jar --input ./build/libs/ --app-version "+jpackage_version+" --name EduMIPS64 --description \"Educational MIPS64 CPU Simulator\" --vendor \"EduMIPS64 Development Team\" --copyright \"Copyright "+LocalDateTime.now().year+", EduMIPS64 development Team\" --license-file ./LICENSE --win-shortcut --win-dir-chooser --win-menu --type msi --icon ./src/main/resources/images/ico.ico --win-per-user-install")

```

Comment:
Instead of creating an intermediate `.bat` file, could we directly invoke `jpackage.exe` from the Gradle task itself?

Reply:
For some reason the jpackage.exe command is not recognized if called directly from the gradle tasks. Probably it doesn't import the env variables. 
I will try other approach

Reply:
I can't seem to get it done. Already tried to specify the full path and still does not work. 
I can add  
```java
    catch (ex:Exception){
        script.delete();
    }
    doLast{
        script.delete();
    }
```
to erase the script file once it is no longer needed. (a finally block should work but doesn't)
This way the file is deleted either the build is successful or not.
Is the best i can do.

Reply:
That's ok, no problem at all. You can leave it as it is and we may investigate it later, for now it works and it's a great new feature!